src/Microsoft.Framework.Configuration.Abstractions/IConfigurationRoot.cs

```diff
@@ -0,0 +1,10 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.Framework.Configuration
+{
+    public interface IConfigurationRoot : IConfiguration
```

Comment:
Not to randomize too much, but it looks like the main reason to split IConfigurationRoot and IConfigurationSection is for Reload?  Does it not make sense to be able to reload the entire configuration object from any node?  I'm just trying to understand what the motivation was to split this out